### PR TITLE
First implementation of workers

### PR DESCRIPTION
**Pull request by kinnou02**
- [x] update requirements
- [x] add scheduler
- [x] improve log

### CODE REVIEW

File: kirin/gtfs_rt/tasks.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+@celery.task(bind=True)
+def gtfs_poller(self, config):
+    logger = logging.getLogger(__name__)
+    logger.info('polling of %s', config['feed_url'])
+    response = requests.get(config['feed_url'])
+    if response.status_code != 200:
+        raise InvalidFeed()
+
+    nav = navitia_wrapper.Navitia(url=config['navitia_url'], token=config['token'])\
```
Comment (antoine-de) [inline kirin/gtfs_rt/tasks.py]: I think we can move this in the model_maker

File: kirin/helper.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+def make_celery(app):
+    celery_app = celery.Celery(app.import_name, broker=app.config['CELERY_BROKER_URL'])
+    celery_app.conf.update(app.config)
+    TaskBase = celery_app.Task
+
+    class ContextTask(TaskBase):
+        abstract = True
+
+        def __init__(self):
```
Comment (antoine-de) [inline kirin/helper.py]: do we need this ? and if yes, don't we need to call the parent constructor ?

File: kirin/tasks.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    # Flask SQLAlchemy will automatically create new sessions for you from
+    # a scoped session factory, given that we are maintaining the same app
+    # context, this ensures tasks have a fresh session (e.g. session errors
+    # won't propagate across tasks)
+    model.db.session.remove()
+
+
+
+from kirin.gtfs_rt.tasks import gtfs_poller
+@celery.task(bind=True)
```
Comment (antoine-de) [inline kirin/tasks.py]: by curiosity what does this `bind` do ?

File: kirin/gtfs_rt/tasks.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+from google.protobuf.message import DecodeError
+
+
+class InvalidFeed(Exception):
+    pass
+
+@celery.task(bind=True)
+def gtfs_poller(self, config):
+    logger = logging.getLogger(__name__)
+    logger.info('polling of %s', config['feed_url'])
```
Comment (antoine-de) [inline kirin/gtfs_rt/tasks.py]: I don't you said there is more work on the logging, but just to log it (:wink:), I think this should be debug

File: kirin/gtfs_rt/tasks.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        raise InvalidFeed()
+
+    nav = navitia_wrapper.Navitia(url=config['navitia_url'], token=config['token'])\
+                         .instance(config['coverage'])
+    nav.timeout = 5
+
+    proto = gtfs_realtime_pb2.FeedMessage()
+    try:
+        proto.ParseFromString(response.content)
+    except DecodeError:
```
Comment (antoine-de) [inline kirin/gtfs_rt/tasks.py]: we should log this

File: kirin/gtfs_rt/tasks.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+class InvalidFeed(Exception):
+    pass
+
+@celery.task(bind=True)
+def gtfs_poller(self, config):
+    logger = logging.getLogger(__name__)
+    logger.info('polling of %s', config['feed_url'])
+    response = requests.get(config['feed_url'])
+    if response.status_code != 200:
```
Comment (antoine-de) [inline kirin/gtfs_rt/tasks.py]: we should log at least if there is something usefull in the response

File: kirin/gtfs_rt/tasks.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        raise InvalidFeed()
```
Comment (xlqian) [inline kirin/gtfs_rt/tasks.py]: it's more like a `PollFailure` than a `InvalidFeed`, no?

File: kirin/helper.py [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+            pass
+
+        def __call__(self, *args, **kwargs):
+            with app.app_context():
```
Comment (xlqian) [inline kirin/helper.py]: why do you need to context the task? because of the app.config?

Comment (kinnou02) [reply kirin/helper.py]: http://flask.pocoo.org/docs/0.12/patterns/celery/

File: kirin/helper.py (hunk 2 above)
Comment (kinnou02) [reply kirin/helper.py]: http://flask.pocoo.org/docs/0.12/patterns/celery/

File: kirin/tasks.py (hunk 3 above)
Comment (kinnou02) [reply kirin/tasks.py]: this `bind` give us access to the task by the self parameter

File: kirin/helper.py (hunk 2 above)
Comment (antoine-de) [reply kirin/helper.py]: I don't see it in the doc :laughing:

File: kirin/tasks.py (hunk 3 above)
Comment (antoine-de) [reply kirin/tasks.py]: but you don't need self no ?

**Pull request merged**